mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -336,9 +342,10 @@ struct LinearizeVectorExtract final
   matchAndRewrite(vector::ExtractOp extractOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     Type dstTy = getTypeConverter()->convertType(extractOp.getType());
-    assert(!(extractOp.getVector().getType().isScalable() ||
-             cast<VectorType>(dstTy).isScalable()) &&
-           "scalable vectors are not supported.");
+    if (extractOp.getVector().getType().isScalable() ||
+        cast<VectorType>(dstTy).isScalable())
```

Comment:
Actually the `dstTy` here is not always a vector type. It could be a scalar type too. 
e.g., `vector.extract %1 [0, 0]: f32 from vector<1024x1024xf32>`. So, cast<VectorType>(dstTy) may cause the crash.

Reply:
Would you say the following is better?
```cpp
if ( (auto vecDstTy = cast<VectorType>(dstTy) && vecDstTy.isScalable()) || extractOp.getVector().getType().isScalable() )
```

Reply:
What's the update here?

Reply:
The test case is there:
```mlir
%0 = vector.extract %arg0[1,1,1]: f32 from vector<2x8x2xf32>
```
but fixing it can get a bit tricky because right now there are a lot of vector result assumptions (e.g., `isLessThanTargetBitWidth()`, `populateVectorLinearizeShuffleLikeOpsPatterns()`)

Reply:
> Any suggestions to make the fix least invasive are welcomed.

You should disable this pattern when the rank of the output is <= 1.